simpleflow/task.py

```diff
@@ -93,16 +112,27 @@ def execute(self):
         if hasattr(method, "execute"):
             task = method(*self.args, **self.kwargs)
             task.context = self.context
+            if self.middleware:
+                self.pre_post_funcs["simpleflow_middleware_prefunc"](task.context)
+
             result = task.execute()
+
             if hasattr(task, "post_execute"):
                 task.post_execute()
+            if self.middleware:
+                self.pre_post_funcs["simpleflow_middleware_postfunc"](task.context)
```

Comment:
I think it could be a good idea to add `result` to the `postfunc` call. Even if we don't plan on using it now, it can contain useful information

Reply:
Agreed; also, we can pass context in both calls and result here as kwargs to be more future-proof